modules/gke-nodepool/variables.tf

```diff
@@ -62,6 +62,30 @@ variable "name" {
   default     = null
 }
 
+variable "network_config" {
+  description = "Network configuration."
+  type = object({
+    enable_private_nodes = optional(bool)
+    pod_range = optional(object({
+      cidr   = optional(string)
+      create = optional(bool, false)
+      name   = optional(string)
+    }))
+    additional_node_network_configs = optional(list(object({
+      network    = string
+      subnetwork = string
+    })), [])
+    additional_pod_network_config = optional(list(object({
+      subnetwork          = string
+      secondary_pod_range = string
+      max_pods_per_node   = string
+    })), [])
+    total_egress_bandwidth_tier        = optional(string)
+    pod_cidr_overprovisioning_disabled = optional(bool, false)
+  })
```

Comment:
You're not using this field.

Also if possible, leave the default as null here to allow the provider to pick the default value for this field.